src/status_im/ui/screens/wallet/recipient/views.cljs

```diff
@@ -0,0 +1,286 @@
+(ns status-im.ui.screens.wallet.recipient.views
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.i18n :as i18n]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.topbar :as topbar]
+            [quo.core :as quo]
+            [status-im.ui.components.colors :as colors]
+            [status-im.utils.debounce :as debounce]
+            [status-im.ui.components.search-input.view :as search-input]
+            [status-im.ui.screens.wallet.components.views :as components]
+            [status-im.ui.components.icons.vector-icons :as icons]
+            [status-im.ui.components.chat-icon.screen :as chat-icon]
+            [status-im.multiaccounts.core :as multiaccounts]
+            [status-im.utils.utils :as utils]
+            [status-im.ui.components.keyboard-avoid-presentation :as kb-presentation]
+            [status-im.ui.components.toolbar :as toolbar]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.stateofus :as stateofus])
+  (:require-macros [status-im.utils.views :as views]))
+
+(defn- recipient-topbar []
+  [topbar/topbar
+   {:navigation    {:on-press
+                    #(do
+                       (re-frame/dispatch [:wallet/recipient-modal-closed])
+                       (re-frame/dispatch [:search/recipient-filter-changed nil])
+                       (re-frame/dispatch [:set-in [:wallet/prepare-transaction :modal-opened?] false])
+                       (re-frame/dispatch [:navigate-back]))}
+    :modal?        true
+    :border-bottom false
+    :title         (i18n/label :t/recipient)
+    :right-accessories
+    [{:icon                :qr
+      :accessibility-label :scan-contact-code-button
+      :on-press            #(re-frame/dispatch [:wallet.send/qr-scanner
+                                                {:handler :wallet.send/qr-scanner-result}])}]}])
+
+(defonce search-active? (reagent/atom false))
+
+(defn search-input-wrapper []
+  (let [search-filter @(re-frame/subscribe [:search/recipient-filter])]
+    [react/view {:padding-horizontal 16
+                 :padding-vertical   10}
+     [search-input/search-input
+      {:search-active? search-active?
+       :search-filter  search-filter
+       :on-cancel      #(re-frame/dispatch [:search/recipient-filter-changed nil])
+       :on-change      (fn [text]
+                         (re-frame/dispatch [:search/recipient-filter-changed text])
+                         (re-frame/dispatch [:set-in [:contacts/new-identity :state] :searching])
```

Comment:
can we just do this at `:search/recipient-filter-changed`?

Reply:
unfortinatelly no , because we have `:search/recipient-filter-changed` in other places, but we need to set search only here